runtime/vm/createramclass.cpp

```diff
@@ -2871,6 +2861,17 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 		result->classFlags |= J9ClassIsAnonymous;
 	}
 
+
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	if (NULL != result) {
```

Comment:
You can split the check above `if ((NULL != result) && (0 != (J9_FINDCLASS_FLAG_ANON & options))) {` and add this code there rather than NULL checking twice